--- faad2-2.7/plugins/mpeg4ip/faad2.cpp.orig	2012-12-09 09:30:54.613929786 +0100
+++ faad2-2.7/plugins/mpeg4ip/faad2.cpp	2012-12-09 11:02:12.133815617 +0100
@@ -21,7 +21,7 @@
 #include "faad2.h"
 #include <mpeg4_audio_config.h>
 #include <mpeg4_sdp.h>
-#include <mp4.h>
+#include <mp4v2/mp4v2.h>
 #include <SDL/SDL.h>
 
 #define DEBUG_SYNC 2
--- faad2-2.7/plugins/mpeg4ip/Makefile.am.orig	2003-08-07 19:21:21.000000000 +0200
+++ faad2-2.7/plugins/mpeg4ip/Makefile.am	2012-12-09 11:12:59.277135462 +0100
@@ -10,6 +10,7 @@
 
 faad2_plugin_la_LIBADD = \
 	$(top_builddir)/libfaad/libfaad.la \
+	-lmp4util \
 	-lm
 
 
