--- faad2/libfaad/Makefile.am.orig	2004-11-08 09:06:09.000000000 +0100
+++ faad2/libfaad/Makefile.am	2007-11-04 11:44:26.132686561 +0100
@@ -3,7 +3,7 @@ lib_LTLIBRARIES = libfaad.la
 include_HEADERS = $(top_srcdir)/include/faad.h \
                   $(top_srcdir)/include/neaacdec.h
 
-libfaad_la_LDFLAGS = -lm
+libfaad_la_LIBADD = -lm
 
 libfaad_la_SOURCES = bits.c cfft.c decoder.c drc.c \
 drm_dec.c error.c filtbank.c \
--- faad2/plugins/xmms/src/Makefile.am.orig	2006-09-25 21:52:53.000000000 +0200
+++ faad2/plugins/xmms/src/Makefile.am	2007-11-04 11:45:59.870028348 +0100
@@ -1,7 +1,7 @@
 local_CFLAGS=`$(XMMS_CONFIG) --cflags` -Wall
 local_LDFLAGS=`$(XMMS_CONFIG) --libs`
-libdir = `$(XMMS_CONFIG) --input-plugin-dir`
-lib_LTLIBRARIES = libmp4.la
+plugindir = `$(XMMS_CONFIG) --input-plugin-dir`
+plugin_LTLIBRARIES = libmp4.la
 
 libmp4_la_CFLAGS = $(local_CFLAGS) -Wall \
        -I$(top_srcdir)/include -I$(top_srcdir)/common/mp4ff
