--- faad2/plugins/xmms/src/libmp4.c.orig	2006-08-07 20:47:45.000000000 +0200
+++ faad2/plugins/xmms/src/libmp4.c	2006-08-23 23:50:29.267477000 +0200
@@ -127,7 +127,7 @@
     gchar*	extention;
 
     extention = strrchr(filename, '.');
-    if(extention &&
+    if(extention && *extention &&
        (!strncasecmp(extention, ".mp4", 4) ||	// official extention
 	!strncasecmp(extention, ".m4a", 4) ||	// Apple mp4 extention
 	!strncasecmp(extention, ".aac", 4)	// old MPEG2/4-AAC extention
