--- faad2/plugins/xmms/src/Makefile.am.orig	2006-09-25 21:52:53.000000000 +0200
+++ faad2/plugins/xmms/src/Makefile.am	2007-11-04 11:45:59.870028348 +0100
@@ -1,7 +1,7 @@
 local_CFLAGS=`$(XMMS_CONFIG) --cflags` -Wall
 local_LDFLAGS=`$(XMMS_CONFIG) --libs`
-libdir = `$(XMMS_CONFIG) --input-plugin-dir`
-lib_LTLIBRARIES = libmp4.la
+plugindir = `$(XMMS_CONFIG) --input-plugin-dir`
+plugin_LTLIBRARIES = libmp4.la
 
 libmp4_la_CFLAGS = $(local_CFLAGS) -Wall \
        -I$(top_srcdir)/include -I$(top_srcdir)/common/mp4ff
