--- faad2.warn/plugins/xmms/src/libmp4.c	2003-12-03 19:37:56.000000000 +0100
+++ faad2/plugins/xmms/src/libmp4.c	2004-06-12 13:53:42.364218112 +0200
@@ -112,14 +112,15 @@
     gchar*	extention;
 
     extention = strrchr(filename, '.');
-    if (extention &&
-	!strcasecmp(extention, ".mp4") ||	// official extention
+    if (extention && *extention) {
+	if(!strcasecmp(extention, ".mp4") ||	// official extention
 	!strcasecmp(extention, ".m4a") ||	// Apple mp4 extention
 	!strcasecmp(extention, ".aac")		// old MPEG2/4-AAC extention
 	){
       return (1);
     }
   }
+  }
   return(0);
 }
 
