--- faad2/Makefile.am.orig	2004-02-06 17:04:39.000000000 +0100
+++ faad2/Makefile.am	2004-02-09 22:11:52.533566432 +0100
@@ -3,6 +3,6 @@
 EXTRA_DIST = faad2.spec
 
 rpm:    Makefile
-       make dist
-       $(RPMBUILD) -ta $(PACKAGE)-$(VERSION).tar.gz
-       rm $(PACKAGE)-$(VERSION).tar.gz
+	make dist
+	$(RPMBUILD) -ta $(PACKAGE)-$(VERSION).tar.gz
+	rm $(PACKAGE)-$(VERSION).tar.gz
--- faad2/libfaad/Makefile.am.orig	2004-02-06 14:12:28.000000000 +0100
+++ faad2/libfaad/Makefile.am	2004-02-09 22:43:21.944332336 +0100
@@ -19,6 +19,7 @@
 codebook/hcb_1.h codebook/hcb_2.h codebook/hcb_3.h codebook/hcb_4.h \
 codebook/hcb_5.h codebook/hcb_6.h codebook/hcb_7.h codebook/hcb_8.h \
 codebook/hcb_9.h codebook/hcb_10.h codebook/hcb_11.h codebook/hcb_sf.h
+libfaad_la_LIBADD = -lm
 
 LIBTOOL_DEPS =
 LIBS =
